app/assets/javascripts/visualizations/highvis/histogram.coffee

```diff
@@ -250,120 +241,89 @@ $ ->
-            name: field.fieldName
+            visible: @configs.displayField is fi
+            name: f.fieldName
             type: 'area'
             xAxis: 1
-            legendIndex: fieldIndex
+            legendIndex: fi
 
       drawToolControls: ->
+        ictx =
+          binSize: @configs.binSize
 
-        controls = ""
-
-        # --- #
-        controls +=  '<div id="toolControl" class="vis_controls">'
-
-        controls += "<h3 class='clean_shrink'><a href='#'>Tools:</a></h3>"
-
-        controls += "<div class='outer_control_div'>"
+        octx =
+          id: 'tools-ctrls'
+          title: 'Tools'
+          body: HandlebarsTemplates[hbCtrl('histogram-tools')](ictx)
 
-        controls += "<h4 class='clean_shrink'>Bin Size</h4>"
+        tools = HandlebarsTemplates[hbCtrl('body')](octx)
+        $('#vis-ctrls').append(tools)
 
-        controls += "Automatic : <br>"
-        controls += "<div id='binSizeSlider' style='width:90%;margin-left:5%'></div><br>"
-
-
-        controls += "Manual: <input id='binSizeInput' class='form-control' value='#{@configs.binSize}'></input>"
-
-        controls += '</div></div></div>'
-
-
-        # Write HTML
-        ($ '#controldiv').append controls
+        # Initialize and track the status of this control panel
+        globals.configs.toolsOpen ?= false
+        initCtrlPanel('tools-ctrls', 'toolsOpen')
 
         # Set up slider
-        ($ '#binSizeSlider').slider
-          range: 'min'
-          value: 2.7 - @binNumSug
+        init =
+          value: @configs.binSize
           min: .5
           max: 2.2
           step: .1
-          slide: (event, ui) =>
-            @binNumSug = 2.7 - Number ui.value
-
-            newBinSize = @defaultBinSize()
-
-            if not fpEq newBinSize, @configs.binSize
-              @configs.binSize = newBinSize
-              ($ '#binSizeInput').val "#{@configs.binSize}"
-              @delayedUpdate()
-
-        # Set up accordion
-        globals.configs.toolsOpen ?= 0
-
-        ($ '#toolControl').accordion
-          collapsible:true
-          active:globals.configs.toolsOpen
-
-        ($ '#toolControl > h3').click ->
-          globals.configs.toolsOpen = (globals.configs.toolsOpen + 1) % 2
-
-        ($ "#binSizeInput").keydown (e) =>
-          if e.keyCode == 13
-
-            newBinSize = Number ($ '#binSizeInput').val()
-
-            if isNaN newBinSize
-              ($ "#binSizeInput").errorFlash()
-              return
-
-            if newBinSize <= 0
-              ($ "#binSizeInput").errorFlash()
-              return
-
-            if ((@globalmax - @globalmin) / newBinSize) < @MAX_NUM_BINS
-              @configs.binSize = newBinSize
-              @update()
-            else
-              alert "Entered bin size would result in too many bins."
-
+        $('#bin-size-slider').attr(init)
+        $('#bin-size-slider').on 'input change', (e) =>
+          @binNumSug = 2.7 - Number(e.target.value)
```

Comment:
Honestly I don't know. Skittles did it, and I think it has to do with positioning the slider.